xla/service/algebraic_simplifier.cc

```diff
@@ -519,6 +484,53 @@ int64_t GetReduceFlops(const HloInstruction* reduce) {
 
 }  // namespace
 
+bool AlgebraicSimplifierVisitor::IsPositive(
+    const HloInstruction* hlo, const AlgebraicSimplifierOptions& options) {
+  // Utility only handles real types.
+  if (IsAnyOperandComplex(hlo)) {
+    return false;
+  }
+  switch (hlo->opcode()) {
+    case HloOpcode::kGetTupleElement: {
+      const HloInstruction* gte_operand = hlo->operand(0);
+      switch (gte_operand->opcode()) {
+        case HloOpcode::kCustomCall: {
+          const auto& target = gte_operand->custom_call_target();
+          return target ==
+                     options.get_cudnn_batchnorm_forward_training_metadata() &&
+                 hlo->tuple_index() == 2;
+        }
+        default:
+          return false;
+      }
+    }
+    case HloOpcode::kExp:
+      return true;
+    case HloOpcode::kPower:
+    case HloOpcode::kAbs:
+    case HloOpcode::kRsqrt:
+    case HloOpcode::kSqrt:
+      return IsPositive(hlo->operand(0), options);
+
+    case HloOpcode::kMultiply: {
+      if (hlo->operand(0) == hlo->operand(1)) {
+        return IsPositive(hlo->operand(0), options);
+      }
+      if (IsPositive(hlo->operand(0), options) &&
+          IsPositive(hlo->operand(1), options)) {
+        return true;
+      }
+      if (!IsNonNegative(hlo->operand(0), options) &&
+          !IsNonNegative(hlo->operand(1), options)) {
+        return true;
+      }
```

Comment:
Remove this case. If either the LHS or RHS is zero, the output will be zero, not positive.